snykTask/src/index.ts

```diff
@@ -146,21 +153,39 @@ async function runSnykTest(
 
   const snykTestToolRunner = tl
     .tool(snykPath)
+    .argIf(taskArgs.testType == TestType.CODE, 'code')
+    .argIf(
+      taskArgs.dockerImageName || taskArgs.testType == TestType.CONTAINER_IMAGE,
+      'container',
+    )
     .arg('test')
     .argIf(
-      taskArgs.severityThreshold,
+      taskArgs.testType != TestType.CODE && taskArgs.severityThreshold,
       `--severity-threshold=${taskArgs.severityThreshold}`,
     )
-    .argIf(taskArgs.dockerImageName, `--docker`)
+    .argIf(
+      taskArgs.testType == TestType.CODE && taskArgs.codeSeverityThreshold,
+      `--severity-threshold=${taskArgs.codeSeverityThreshold}`,
+    )
     .argIf(taskArgs.dockerImageName, `${taskArgs.dockerImageName}`)
     .argIf(fileArg, `--file=${fileArg}`)
     .argIf(taskArgs.ignoreUnknownCA, `--insecure`)
     .argIf(taskArgs.organization, `--org=${taskArgs.organization}`)
     .argIf(taskArgs.projectName, `--project-name=${projectNameArg}`)
     .arg(`--json-file-output=${jsonReportOutputPath}`)
-    .argIf(isDebugMode(), '-d')
+    .argIf(isDebugMode() && taskArgs.testType != TestType.CODE, '-d')
+    .argIf(taskArgs.testType == TestType.CODE, `--json`)
     .line(taskArgs.additionalArguments);
 
+  // pipe to json file output with --json to handle no code issues found
+  if (taskArgs.testType == TestType.CODE) {
```

Comment:
suggestion: handle this like the other products. If not possible, why?

Reply:
because snyk code test does not generate a json file for an no-issues-found scan. The toolrunner provides a method to pipe the console output to a file (json) so that it could be converted to html.